sway-ir/src/variable.rs

```diff
@@ -67,3 +67,64 @@ impl LocalVar {
         context.local_vars[self.0].mutable = mutable;
     }
 }
+
+/// A wrapper around an [ECS](https://github.com/orlp/slotmap) handle into the
+/// [`Context`].
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, DebugWithContext)]
+pub struct GlobalVar(#[in_context(global_vars)] pub slotmap::DefaultKey);
+
+#[doc(hidden)]
+#[derive(Clone, DebugWithContext)]
+pub struct GlobalVarContent {
+    pub ptr_ty: Type,
+    pub initializer: Option<Constant>,
+    pub mutable: bool,
```

Comment:
`mutable` is currently nowhere used. It's just a marker. Considering that we've decided to support `GlobalVar`s in the IR, and not `GlobalConst`s we should also check the semantics around mutability.

In the verifier:
- a non-mutable `GlobalVar` must be initialized.
- only mutable `GlobalVar` can be stored to.

In the printer and parser we also need support for mutability and optional initilizer, but this I've already commented in respected files.

Sadly, also for the mutability of `LocalVar`s we don't have any checks, it's just a marker.

Maybe to have a separate PR to enforce mutability semantics for both `Local` and `Global` vars? But for the printer and parser at least I see it as a part of this PR.

Reply:
> a non-mutable GlobalVar must be initialized.

I'll add a check for this.

> only mutable GlobalVar can be stored to.

This isn't easy without an analysis (we will need to use something like `get_sym` to get to the symbol being stored to in a `Store`. I'll skip this part for later.

Reply:
Ok, turns out, when we create locals, at many places, we do this: `new_local_var(context, temp_arg_name, value_type, None, false)`, where we aren't providing an initializer, but marking it immutable. So my verification check added for locals fails. I'll remove it for now, to fix the locals issue separately.